src/main/java/com/microsoft/sqlserver/jdbc/PerformanceLogCallback.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+/**
+ * Callback interface for publishing performance logs.
+ */
+public interface PerformanceLogCallback {
+
+    /**
+     * Publish a performance log entry.
+     *
+     * @param durationMs      The duration of the operation in milliseconds.
+     * @param activity        The type of activity being logged.
+     * @param exceptionMessage An exception message, if an error occurred.
+     */
+    void publish(long durationMs, PerformanceActivity activity, String exceptionMessage);
+

```

Comment:
Consider changing the parameter from String exceptionMessage to Exception exception. Passing the full Exception object provides richer context (stack traces, exception types, cause chains) for better debugging and observability rather than just the string message, allowing implementations to make more informed decisions about error handling.

void publish(long durationMs, PerformanceActivity activity, Exception exception) throws Exception;